src/main/java/seedu/bookmark/model/ModelManager.java

```diff
@@ -142,9 +173,62 @@ public void updateFilteredBookList(Predicate<Book> predicate) {
 
     @Override
     public void sortFilteredBookList(Comparator<Book> comparator) {
+        if (comparator instanceof BookNameComparator) {
+            setSortingPreference(PREFIX_NAME.toString());
+        } else if (comparator instanceof BookGenreComparator) {
+            setSortingPreference(PREFIX_GENRE.toString());
+        } else if (comparator instanceof BookPagesReadComparator) {
+            setSortingPreference(PREFIX_BOOKMARK.toString());
+        } else if (comparator instanceof BookReadingProgressComparator) {
+            setSortingPreference(PREFIX_READING_PROGRESS.toString());
+        }
+        this.comparator = comparator;
         this.library.sortBooks(comparator);
     }
 
+    /**
+     * Returns a comparator based on the input prefix
+     * @return Comparator based on input prefix
+     */
+    public Comparator<Book> comparatorGenerator(Prefix inputPrefix) {
+        Comparator<Book> comparator = null;
+        if (inputPrefix == PREFIX_NAME) {
+            comparator = new BookNameComparator();
+        } else if (inputPrefix == PREFIX_GENRE) {
+            comparator = new BookGenreComparator();
+        } else if (inputPrefix == PREFIX_BOOKMARK) {
+            comparator = new BookPagesReadComparator();
+        } else if (inputPrefix == PREFIX_READING_PROGRESS) {
+            comparator = new BookReadingProgressComparator();
+        }
+        return comparator;
+    }
+
+    /**
+     * Returns a prefix based on the input string.
+     * @return Prefix based on the input string.
+     */
+    public Prefix prefixGenerator(String prefix) {
```

Comment:
I don't think this method belongs here, maybe define it as a utility method in `CliSyntax` ?